### PR TITLE
Cancel scheduled frame when unmounting useTransform

### PR DESCRIPTION
**Pull request by mattgperry**
This PR ensures that when` useTransform`'s subscription effect is cleaned up it cancels any scheduled animation frame.

 I haven't been able to capture this in a test but have verified it fixes the corresponding bug in our Slideshow component.

### CODE REVIEW

File: dev/package.json [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
     "scripts": {
         "dev": "webpack serve --config ./webpack/config.js --hot"
     },
     "dependencies": {
         "@react-three/drei": "^7.27.3",
         "@react-three/fiber": "^8.2.2",
-        "framer-motion": "^7.5.0",
-        "framer-motion-3d": "^7.5.0",
+        "framer-motion": "^7.5.1-beta.0",
+        "framer-motion-3d": "^7.5.1-beta.0",
```
Comment (nvh) [inline dev/package.json:12]: should we include beta versions here?

File: packages/framer-motion/src/value/use-on-change.ts [hunk 2]
```diff
@@ -11,9 +11,17 @@ export function useOnChange<T>(
     }, [callback])
 }
 
-export function useMultiOnChange(values: MotionValue[], handler: () => void) {
+export function useMultiOnChange(
+    values: MotionValue[],
+    handler: () => void,
+    clean: () => void
```
Comment (nvh) [inline packages/framer-motion/src/value/use-on-change.ts:17]:
```suggestion
    cleanup: () => void
```

**Review by nvh — APPROVED** [review]
There's an empty test file added, we should remove that.

File: dev/package.json (hunk 1 above)
Comment (mattgperry) [reply dev/package.json:12]: this is just cruft really, it'll get updated to 7.5.1 when I publish from main

File: packages/framer-motion/src/value/use-on-change.ts (hunk 2 above)
Comment (nvh) [reply packages/framer-motion/src/value/use-on-change.ts:17]: Shouldn't we also make this optional?

**Pull request merged**